perf/README

```diff
@@ -110,19 +110,24 @@ It does 100000 repetitions divided evenly among each thread. The number of
 threads to use is provided as an argument and the test reports the average time
 take to execute a block of 1000 OSSL_PROVIDER_do_all() calls.
 
-pemread
--------------
-
-The pemread test repeatedly calls the PEM_read_bio_PrivateKey() function on
-a memory BIO with a private RSA key. It does 100000 repetitions divided evenly
-among each thread. The number of threads to use is provided as an argument and
-the test reports the average time take to execute a block of 1000
-PEM_read_bio_PrivateKey() calls.
-
 rwlocks
 -------------
 the rwlocks test creates the command line specified number of threads, splitting
 them evenly between read and write functions (though this is adjustable via the
 LOCK_WRITERS environment variable).  Threads then iteratively acquire a shared
 rwlock to read or update some shared data.  The number of read and write
 lock/unlock pairs are reported as a performance measurement
+
+pkeyread
+--------
+
+The pkeyread test repeatedly calls the PEM_read_bio_PrivateKey() function on
+a memory BIO with a private RSA key, when it is running in pem mode (-f pem).
+If test is running in der mode (-f der) it calls to d2i_PKCS8_PRIV_KEY_INFO_bio()
```

Comment:
IMO this is wrong. Testing d2i_PKCS8_PRIV_KEY_INFO_bio() is not that much useful. That function does not invoke decoders at all. Instead d2i_PrivateKey_ex() on the same data would provide much more useful results.

However I see that you then call EVP_PKCS82PKEY() - this is the actual function where the decoders are invoked. Funnily enough it first reverts the d2i_PKCS8_PRIV_KEY_INFO operation by calling i2d which is then passed to a decoder.

IMO to see more realistic performance of DER decoding of a PKEY d2i_PrivateKey_ex() should be tested instead of these two calls together.